### DISCUSSION TITLE
plug proposal: Disk Management

### DISCUSSION THREAD

**Discussion by jepotter1-archive** [elementary/settings#71] [Ideas]
From elementary/installer#324.

There should be a new switchboard plug that allows the user to:
* See what partitions are on their disk and how much space each is taking graphically
* Visualize used vs free disk space
* Boot into recovery mode to edit partition
* Delete elementaryOS

```
                          Hard Drive Layout
|-----elementary-----|-------------Ubuntu-----|
         32 GB                          103 GB
[Reboot into Recovery Mode] [Remove elementaryOS]
```

**Comment by danirabbit** [comment]
I think a utility like this is outside the scope of Switchboard. I would recommend installing GNOME Disks from AppCenter.

Would love to hear any other opinions from @elementary/ux but I think for now I'll close this as "Out of scope"

**Comment by cassidyjames** [comment]
@danrabbit I can see some argument for managing disks/storage in System Settings. In other operating systems I can check somewhere in the system to see what's using my storage (with something like a Granite storage bar), a way to free up space, and possibly a way to format new partitions (i.e. when adding a new hard drive to an existing machine).

I don't think it's an urgent need since third-party tools exist, but I do think having it integrated into the system a bit more would be nice and sane.

**Comment by danirabbit** [comment]
Hm yeah I think I'm worried about a runaway scope here for Switchboard, but I'll reopen for the sake of discussion

**Comment by megatux** [comment]
Also, some people ask for ways to handling automatic mounting at startup. (e.g. in Slack channel) and using a `gio mount /dev/whatever` at startup is not a very user friendly thing to do.
Adding a link to a related [discussion in Files](https://github.com/elementary/files/issues/729).

**Comment by cassidyjames** [comment]
@mmstick I was just thinking… would using distinst here make any sense for a disk formatting utility, or would it make more sense to use the underlying libraries more directly?

**Comment by Justinzobel** [comment]
An initial concept seems to have been made here:
https://bazaar.launchpad.net/~optimisme/+junk/switchboard-drives/files

It's based on this mockup:
![d5fgg7y-abca3ebe-74d7-45ec-86f5-3d76776e4bc7](https://user-images.githubusercontent.com/1370117/65368167-a1326280-dc7c-11e9-8d35-5536469961e9.png)

If anyone wants a base to start from. The code doesn't compile, it calls for a package that I'm guessing has been renamed or obsoleted:

```
Command-line option `--thread` is deprecated and will be ignored
error: Package `pantheon' not found in specified Vala API directories or GObject-Introspection GIR directories
```
